python-client/giskard/push/contribution.py

```diff
@@ -66,34 +75,73 @@ def create_contribution_push(model: BaseModel, ds: Dataset, df: pd.DataFrame) ->
                 model_type=model.meta.model_type,
                 correct_prediction=correct_prediction,
             )
+        elif _most_important_feature_is_text or _text_is_the_only_feature:
```

Comment:
could you extract every "if" branch to a separate method since Sonar is complaining anyway